pkg/api/graph/graph.go

```diff
@@ -349,12 +368,24 @@ func (g Graph) addEdges(edges []graph.Edge, fn EdgeFunc) {
 	}
 }
 
+// RemoveEdge removes the provided edge from the graph
+func (g Graph) RemoveEdge(edge graph.Edge) {
```

Comment:
I'm not sold on this over subgraph, but I'll keep reading.  Mutating a graph while keeping its meaning the same isn't always obvious.  This could end up being semantically dangerous to accurate edge analysis while looking pretty innocuous.